internal/api/v1/services/instance.go

```diff
@@ -255,44 +258,81 @@ func (s *InstanceService) handleInfrastructureDeletion(
 		return fmt.Errorf("no instances found for job %d", job.ID)
 	}
 
-	// Calculate how many instances to delete
-	numberOfInstancesToDelete := 0
+	// TODO: this section of code could be refactored to be unit tested
+	// Check if we have specific instance names to delete
+	var instancesToDelete []models.Instance
+	var specificNamesToDelete []string
+
+	// Collect specific instance names to delete
 	for _, instanceReq := range infraReq.Instances {
-		numberOfInstancesToDelete += instanceReq.NumberOfInstances
+		if instanceReq.Name != "" {
+			specificNamesToDelete = append(specificNamesToDelete, instanceReq.Name)
+		}
 	}
 
-	if numberOfInstancesToDelete <= 0 {
-		return fmt.Errorf("invalid number of instances to delete: %d", numberOfInstancesToDelete)
-	}
+	// If specific names are provided, find those instances
+	if len(specificNamesToDelete) > 0 {
+		fmt.Printf("ℹ️ Deleting specific instances by name: %v\n", specificNamesToDelete)
+
+		// Find instances with the specified names
+		for _, name := range specificNamesToDelete {
+			found := false
+			for _, instance := range instances {
+				if instance.Name == name {
+					instancesToDelete = append(instancesToDelete, instance)
+					found = true
+					break
+				}
+			}
+			if !found {
+				fmt.Printf("⚠️ Warning: Instance with name '%s' not found\n", name)
+			}
+		}
+
+		if len(instancesToDelete) == 0 {
+			return fmt.Errorf("none of the specified instances were found")
+		}
```

Comment:
This part is confusing, it receives instances to delete from the input, looks for non empty ones then searches them inside the same request to see if they are there. these seem unnecessary as the input is coming from the user. it is suggested to look it up in the DB with OwnerID filter (does not exist currently for instances) and then delete the ones that pass it. Then update the DB. I know that is out of scope of this PR, so let's work on it in a followup PR.

Reply:
agreed. more clean up needed.

Reply:
https://github.com/celestiaorg/talis/issues/97